### DISCUSSION TITLE
Newbie: how to use openUrl

### DISCUSSION THREAD

**Discussion by TriXxieDK** [elgatosf/elgato#22] [Help]
Hi
I'm new to this, but trying to modify an existing plugin to my needs. I want to call openUrl, to make StreamDeck App open an URL in default browser. Im plugins main method I'm doing this:
`    $SD.api.send({
        "event": "openUrl",
        "payload": {
            "url": "https://www.elgato.com"
        }
    });
`
... but nothing happens. Does'nt seem to break syntax tho, as rest of plugin executes flawlessly. It's obviously not on plugin load that I want to open a URL, but so far I'm just testing, and I'll move it to the right place in code when I get it working.

**Comment by ChekTek** [comment]
Hi!

We recommend you use the quick start guide in this template repository to get started: https://github.com/elgatosf/streamdeck-plugin-template

Best,
Zack

**Comment by TriXxieDK** [comment]
I've litterally spent hours, trying all sorts of combinations, reading the JS files, and still coming up with nothing that works. According to api.js I should be able to call `$SD.api.openUrl("https://www.elgato.com");` but when I do, nothing. I've also tried `$SD.api.send(context, "openUrl", { "url": "https://www.elgato.com" } );` aaand nothing. Starting to get quite frustrated by this, as it should be the simplest task in the world :'(

**Comment by ChekTek** [comment]
If you follow the getting started guide in the template above, you should be able to call `$SD.openUrl('https://www.elgato.com');` once you receive the event letting you know that Stream Deck is connected to the plugin.
